languages/swift/src/main/java/de/jplag/swift/SwiftTokenType.java

```diff
@@ -0,0 +1,63 @@
+package de.jplag.swift;
+
+import de.jplag.TokenType;
+
+public enum SwiftTokenType implements TokenType {
+    IMPORT("IMPORT"),
+    CLASS_DECLARATION("CLASS"),
+    STRUCT_DECLARATION("STRUCT"),
+    ENUM_DECLARATION("ENUM"),
+    PROTOCOL_DECLARATION("PROTOCOL"),
+    CLASS_BODY_BEGIN("CLASS{"),
+    CLASS_BODY_END("}CLASS"),
+    STRUCT_BODY_BEGIN("STRUCT{"),
+    STRUCT_BODY_END("}STRUCT"),
+    ENUM_BODY_BEGIN("ENUM{"),
+    ENUM_BODY_END("}ENUM"),
+    ENUM_LITERAL("ENUM CASE"),
+    PROTOCOL_BODY_BEGIN("PROTOCOL{"),
+    PROTOCOL_BODY_END("}PROTOCOL"),
+    PROPERTY_DECLARATION("PROPERTY"),
+    PROPERTY_ACCESSOR_BEGIN("ACCESSOR{"),
+    PROPERTY_ACCESSOR_END("}ACCESSOR"),
+    FUNCTION("FUNCTION"),
+    FUNCTION_PARAMETER("PARAMETER"),
+    FUNCTION_BODY_BEGIN("FUNCTION{"),
+    FUNCTION_BODY_END("}FUNCTION"),
+    CLOSURE_BODY_BEGIN("CLOSURE{"),
+    CLOSURE_BODY_END("}CLOSURE"),
+    FOR_BODY_BEGIN("FOR{"),
+    FOR_BODY_END("}FOR"),
+    IF_BODY_BEGIN("IF{"),
+    IF_BODY_END("}IF"),
+    SWITCH_BODY_BEGIN("SWITCH{"),
+    SWITCH_BODY_END("}SWITCH"),
+    SWITCH_CASE("CASE"),
+    WHILE_BODY_BEGIN("WHILE{"),
+    WHILE_BODY_END("}WHILE"),
+    REPEAT_WHILE_BODY_BEGIN("REPEAT{"),
+    REPEAT_WHILE_BODY_END("}REPEAT"),
```

Comment:
I would suggest mapping the different loop types to the same tokens in order to reduce the impact of obfuscation attacks. What do you think?

Reply:
The Swift token types follow quite closely the Java token types design (or rather any object-oriented language token types design). Thus, I also added the different token types for each loop / branch. 
We might refactor them to just `BRANCH_BODY_BEGIN/_END` and `LOOP_BODY_BEGIN/_END`. For the loops, this refactoring might work. For the branching, I see some problems:

```Swift
if (true) { /* ... */ }
//        |BRANCH_BODY_BEGIN
//                    |BRANCH_BODY_END
else { /* ... */ }
//   |BRANCH_BODY_BEGIN
//               |BRANCH_BODY_END

switch (value) {
case 1: /* ... */
//      |BRANCH_BODY_BEGIN
//              |BRANCH_BODY_END
default: /* ... */
//      |BRANCH_BODY_BEGIN
//              |BRANCH_BODY_END
}
// until here, everything's fine...

switch (value) {
case 3: fallthrough // executes code in next case block
// should this generate tokens or be treated as the same branch block as next case?
case 4, 5: break // i.e. no-op
// should this generate tokens? Would be not present in if-else conditions
}
```

It is no effort refactoring the language frontend to use unified branch / loop token types, and simply leaving out the special cases (i.e. the cases above each would generate tokens). However, I am not sure whether we want this (right now), as it would be inconsistent with the design of other object-oriented languages. It might be beneficial to keep the frontends consistent (to some extent) and perform a study on OO languages to assess the effect of such unified loop / branch tokens and then refactor all OO language frontends together (in case the study results indicate unified token types are to be preferred).

But these are only considerations. If you have a strong opinion towards these unified branch / loop token types, I have no problem with refactoring it.

Reply:
Good points, we can leave it for now. In Java, it seems that currently only switches map to the same tokens.